### PR TITLE
eip-1559: add more constraints to the tx validation

### PR DESCRIPTION
**Pull request by holiman**
Follow-up from the discussions on https://github.com/ethereum/pm/issues/321#issuecomment-850230251 

cc @MicahZoltu @vbuterin

### CODE REVIEW

**Comment by alita-moore** [comment]
Hi! I'm a bot, and I wanted to automerge your PR, but couldn't because of the following issue(s):


	 - eip-1559.md requires approval from one of (vbuterin, econoar, afdudley, mslipper, i-norden, abdelhamidbakhta)
 @vbuterin @econoar @afdudley @mslipper @i-norden @abdelhamidbakhta

File: EIPS/eip-1559.md [hunk 1]
```diff
@@ -209,6 +209,13 @@ class World(ABC):
 
 			# ensure that the user was willing to at least pay the base fee
 			assert transaction.max_fee_per_gas >= block.base_fee_per_gas
+
+			# The first two of these four rules are implicit due to the next two rules
+			assert transaction.max_fee_per_gas < 2**256           # Prevent impossibly large numbers
+			assert transaction.max_priority_fee_per_gas < 2**256  # Prevent impossibly large numbers
+			assert transaction.max_fee_per_gas >= transaction.max_priority_fee_per_gas      # The total must be the larger of the two
+			assert sender.balance >= gasLimit * transaction.max_fee_per_gas
```
Comment (RatanRSur) [inline EIPS/eip-1559.md:217]: sender or signer?

Comment (MicahZoltu) [reply EIPS/eip-1559.md:217]: `signer`.  That is my bad, I didn't expect my hacked together sample code in Discord to make it into the spec.  😜

Comment (chfast) [reply EIPS/eip-1559.md:217]: This is weird when you specify the expected value for an assert on the left side of a logical operator.

Comment (MicahZoltu) [reply EIPS/eip-1559.md:217]: I consider the right side to be the expected value here.  We are checking to make sure the sender has enough money to cover costs, so the "sender's balance" is the actual, and `gasLimit * transaction.max_fee_per_gas` is the expected.

**Pull request merged**